src/modules/70-pipeline/pages/execution/ExecutionArtifactsView/ExecutionArtifactsView.tsx

```diff
@@ -108,6 +109,15 @@ export const getArtifacts: (stages: ExecutionNode[]) => Artifact[] = stages => {
         node
       })) || []
 
+    // CD stage
+    if (node.stepDetails && !isEmpty(node.stepDetails)) {
```

Comment:
Nit: Isn't `!isEmpty(node.stepDetails)` enough?
```suggestion
    if (!isEmpty(node.stepDetails)) {
```

Reply:
`isEmpty(node.stepDetails)` check alone further complains on types (within the block node.stepDetails is still undefined)